src/mudlet-lua/lua/generic-mapper/generic_mapper.xml

```diff
@@ -0,0 +1,2751 @@
+    local statuses = {}
+    local doors = getDoors(roomID)
+    local dir
+    for k,v in pairs(exits) do
+        dir = short[k] or short[v]
+        if table.contains({'u','d'},dir) then
+            dir = exitmap[dir]
+        end
+        if not doors[dir] or doors[dir] == 0 then
+            return false
+        else
+            statuses[dir] = doors[dir]
+        end
+    end
+    return statuses
+end
+
+local function find_room(name, area)
+    -- looks for rooms with a particular name, and if given, in a specific area
+    local rooms = searchRoom(name)
+    if type(area) == "string" then
+        local areas = getAreaTable() or {}
+        for k,v in pairs(areas) do
+            if string.lower(k) == string.lower(area) then
+                area = v
+                break
+            end
+        end
+        area = areas[area] or nil
+    end
+    for k,v in pairs(rooms) do
+        if string.lower(v) ~= string.lower(name) then
+            rooms[k] = nil
+        elseif area and getRoomArea(k) ~= area then
+            rooms[k] = nil
+        end
+    end
+    return rooms
+end
+
+local function getRoomStubs(roomID)
+    -- turns stub info into table similar to exit table
+    local stubs = getExitStubs(roomID)
+    if type(stubs) ~= "table" then stubs = {} end
+    -- check handling of custom exits here
+    local tmp
+    for i = 13,#stubmap do
+        tmp = tonumber(getRoomUserData(roomID,"stub"..stubmap[i]))
+        if tmp then table.insert(stubs,tmp) end
+    end
+
+    local exits = {}
+    for k,v in pairs(stubs) do
+        exits[stubmap[v]] = 0
+    end
+    return exits
+end
+
+local function connect_rooms(ID1, ID2, dir1, dir2, no_check)
+    -- makes a connection between rooms
+    -- can make backwards connection without a check
+    local match = false
+    if not ID1 and ID2 and dir1 then
+        error("Connect Rooms: Missing Required Arguments.",2)
+    end
+    dir2 = dir2 or reverse_dirs[dir1]
+    -- check handling of custom exits here
+    if stubmap[dir1] &lt;= 12 then
+        setExit(ID1,ID2,stubmap[dir1])
+    else
+        setRoomUserData(ID1,"exit " .. dir1,ID2)
+    end
+    if stubmap[dir1] &gt; 13 then
+        -- check handling of custom exits here
+        setRoomUserData(ID1,"stub"..dir1,"")
+    end
+    local doors1, doors2 = getDoors(ID1), getDoors(ID2)
+    local dstatus1, dstatus2 = doors1[short[dir1]] or doors1[dir1], doors2[short[dir2]] or doors2[dir2]
+    if dstatus1 ~= dstatus2 then
+        if not dstatus1 then
+            add_door(ID1,dir1,dstatus2)
+        elseif not dstatus2 then
+            add_door(ID2,dir2,dstatus1)
+        end
+    end
+    if map.mode ~= "complex" then
+        local stubs = getRoomStubs(ID2)
+        if stubs[dir2] then match = true end
+        if (match or no_check) then
+            -- check handling of custom exits here
+            if stubmap[dir1] &lt;= 12 then
+                setExit(ID2,ID1,stubmap
```

Comment:
It would be more intuitive for the user if the script said what to do:

> show_err("What area name would you like to use? <yellow>start mapping <area name><reset>")

Reply:
You are quoting a random bit of code here, but I assume this is when they try to "start mapping", and it has no area name to work with?

Reply:
Github is showing it buggy - have a look at https://github.com/Mudlet/Mudlet/pull/1940/files/96086904f7fa76e40d49c0f9ac28e7312e9e358c#diff-72c499bbe5bf5b3d55b552488d687aaaR2023

Reply:
Oh hey... old error message code. I'll fix that in passing.